### PR TITLE
Avoid attribute protocol

### PR DESCRIPTION
**Pull request by cthulahoops**
First batch of my big change (breaking it up a bit for review friendliness).

### CODE REVIEW

**Comment by cthulahoops** [comment]
All this use of AttributeProtocol is using the internal interface rather than the proper vm attribute path.

File: src/main.rs [hunk 1]
```diff
@@ -176,8 +170,9 @@ fn run_shell(vm: &mut VirtualMachine) -> PyResult {
         println!("No previous history.");
     }
 
-    let ps1 = &objstr::get_value(&vm.sys_module.get_attr("ps1").unwrap());
-    let ps2 = &objstr::get_value(&vm.sys_module.get_attr("ps2").unwrap());
+    let sys_module = vm.sys_module.clone();
+    let ps1 = &objstr::get_value(&vm.get_attribute(sys_module.clone(), "ps1").unwrap());
+    let ps2 = &objstr::get_value(&vm.get_attribute(sys_module, "ps2").unwrap());
```
Comment (coolreader18) [inline src/main.rs]: CPython uses an empty string if somebody deletes `sys.ps{1,2}`, we could probably do the same.

File: vm/src/frame.rs [hunk 2]
```diff
@@ -810,13 +810,13 @@ impl Frame {
         // If we're importing a symbol, look it up and use it, otherwise construct a module and return
         // that
         let obj = match symbol {
-            Some(symbol) => module.get_attr(symbol).map_or_else(
-                || {
+            Some(symbol) => match vm.get_attribute(module, symbol.as_str()) {
+                Ok(obj) => Ok(obj),
+                Err(_) => {
```
Comment (coolreader18) [inline vm/src/frame.rs]: You could just use `.map_err()` here.

File: vm/src/import.rs [hunk 3]
```diff
@@ -63,8 +64,13 @@ pub fn import_module(
...
 
-fn find_source(vm: &VirtualMachine, current_path: PathBuf, name: &str) -> Result<PathBuf, String> {
-    let sys_path = vm.sys_module.get_attr("path").unwrap();
+fn find_source(
+    vm: &mut VirtualMachine,
+    current_path: PathBuf,
+    name: &str,
+) -> Result<PathBuf, String> {
+    let sys_module = &vm.sys_module;
+    let sys_path = vm.get_attribute(sys_module.clone(), "path").unwrap();
```
Comment (coolreader18) [inline vm/src/import.rs]: This applies to all these changes, but why not just do `vm.get_attribute(vm.sys_module.clone(), ...)`?

Comment (cthulahoops) [reply vm/src/import.rs]: I thought I was running into an immutable borrow while borrowed mutably issue (the one you run into with ```vm.do_something(vm.something())```), but that doesn't happen with a simple member access.

Comment (cthulahoops) [reply vm/src/import.rs]: Fixed.

File: vm/src/frame.rs (hunk 2 above)
Comment (cthulahoops) [reply vm/src/frame.rs]: Yes, that's neater.

**Comment by cthulahoops** [comment]
Review comments addressed. Merging.

**Pull request merged**